lib/puppet/functions/foreman/smartvar.rb

```diff
@@ -0,0 +1,38 @@
+# Query Foreman in order to resolv a smart variable
+# Foreman holds all the value names and their possible values,
+# this function simply ask foreman for the right value for this host.
+
+require "net/http"
+require "net/https"
+require "uri"
+require "timeout"
+
+Puppet::Functions.create_function(:'foreman::smartvar') do
+  dispatch :smartvar do
+    required_param 'String[1]', :var
+    optional_param 'String', :foreman_url
```

Comment:
There is a [Stdlib::Httpurl](https://github.com/puppetlabs/puppetlabs-stdlib#stdlibhttpurl)